intro-HPC/ch_running_batch_jobs.tex

```diff
@@ -40,6 +40,331 @@ \chapter{Running batch jobs}
+mathematical software such as MATLAB and SAS, as well as other applications and
+libraries.
+
+In order to administer the active software and their environment variables, a
+\strong{\emph{``module''}} package has been developed, which:
+
+\begin{enumerate}
+\item  Activates or deactivates \underbar{software packages} and their dependencies.
+\item  Allows setting and unsetting of \underbar{environment variables}, including adding and deleting entries from database-type environment variables.
+\item  Does this in a \underbar{shell-independent} fashion (necessary information is stored in the accompanying module configuration file).
+\item  Takes care of \underbar{versioning aspects}: For many libraries, multiple versions are installed and maintained. The module system also takes care of the versioning of software packages in case multiple versions are installed. For instance, it does not allow multiple versions to be loaded at same time.
+\item  Takes care of \underbar{dependencies}: Another issue arises when one considers library versions and the dependencies they create. Some software requires an older version of a particular library to run correctly (or at all). Hence a variety of version numbers is available for important libraries.
+\end{enumerate}
+
+This is all managed with the ``\emph{module}'' command, which is explained in the next sections.
+
+\subsection{Available modules}
+
+A large number of software packages are installed on the \hpc clusters. A
+list of all currently available software can be obtained by typing:
+
+\begin{prompt}
+%\shellcmd{module av}%
+\end{prompt}
+
+or by typing it in full:
+
+\begin{prompt}
+%\shellcmd{module available}%
+\end{prompt}
+
+This will give some output such as:
+
+\inputsite{available-modules}
+
+This gives a full list of software packages that can be loaded. Note that
+modules starting with a capital letter are listed first.
+
+\subsection{Organisation of modules in toolchains}
+
+The amount of modules on the VSC systems can be overwhelming, and it is not
+always immediately clear which modules can be loaded safely together if you
+need to combine multiple programs in a single job to get your work done.
+
+Therefore the VSC has defined so-called \strong{toolchains} on the newer VSC-clusters.
+A toolchain contains a C/C++ and Fortran compiler, a MPI library and some basic
+math libraries for (dense matrix) linear algebra and FFT. Two toolchains are
+defined on most VSC systems. One, the \lstinline|intel| toolchain, consists of the Intel
+compilers, MPI library and math libraries. The other one, the \lstinline|foss| toolchain,
+consists of Open Source components: the GNU compilers, OpenMPI, OpenBLAS and
+the standard LAPACK and ScaLAPACK libraries for the linear algebra operations
+and the FFTW library for FFT. The toolchains are refreshed twice a year,
+which is reflected in their name.
+
+E.g., \lstinline|foss-2014a| is the first version of the \lstinline|foss| toolchain in 2014.
```

Comment:
update to `2018` :), and maybe use `\year` here if something like that exists in LaTeX